src/main/python/org/broadinstitute/hellbender/gatktool/tool.py

```diff
@@ -0,0 +1,175 @@
+"""
+Functions and classes used to extend a GATK tool with Python.
+
+GATK uses two FIFOs to communicate wth Python. The "ack" FIFO is read by GATK
+and written by Python code, and is used to signal that a Python command has
+completed execution. The "data" FIFO is written by GATK and read by Python,
+and is used to pass data to Python from Java.
+
+Most of the functions in this module are intended to be called by GATK via
+the StreamingPythonScriptExecutor Java class, and are not called by Python
+code directly. The one exception is the readDataFIFO function, which can be
+used to read data that had been passed to Python by GATK Java code.
+"""
+
+import sys
+import os
+
+_ackFIFO = None
```

Comment:
If you refactored this as another Python class, you could get rid of these globals.  It might make it a little cleaner.

Reply:
These could be removed and instead created by the having the executor create them, but at least this way they explicitly appear in the Python code. Per discussion yesterday I'll leave these as is.